src/mesh/voxel_cloud.rs

```diff
@@ -367,84 +518,186 @@ impl VoxelCloud {
+    voxel_dimensions: Vector3<f32>,
+) -> Point3<i32> {
+    assert!(
+        !approx::relative_eq!(voxel_dimensions.x, 0.0)
+            && !approx::relative_eq!(voxel_dimensions.y, 0.0)
+            && !approx::relative_eq!(voxel_dimensions.z, 0.0),
+        "Voxel dimensions can't be 0.0"
+    );
+    Point3::new(
+        (point.x / voxel_dimensions.x).round() as i32 - block_start.x,
+        (point.y / voxel_dimensions.y).round() as i32 - block_start.y,
+        (point.z / voxel_dimensions.z).round() as i32 - block_start.z,
+    )
+}
+
+/// Calculates the center of a voxel in model-space coordinates from
+/// absolute voxel coordinates (relative to the voxel space origin).
+#[allow(dead_code)]
+fn absolute_voxel_to_cartesian_coords(
+    absolute_coords: &Point3<i32>,
+    voxel_dimensions: Vector3<f32>,
+) -> Point3<f32> {
+    assert!(
+        !approx::relative_eq!(voxel_dimensions.x, 0.0)
+            && !approx::relative_eq!(voxel_dimensions.y, 0.0)
+            && !approx::relative_eq!(voxel_dimensions.z, 0.0),
+        "Voxel dimensions can't be 0.0"
+    );
+    Point3::new(
+        absolute_coords.x as f32 * voxel_dimensions.x,
+        absolute_coords.y as f32 * voxel_dimensions.y,
+        absolute_coords.z as f32 * voxel_dimensions.z,
+    )
+}
+
+/// Calculates the center of a voxel in model-space coordinates from voxel
+/// coordinates relative to the voxel block start.
+fn relative_voxel_to_cartesian_coords(
+    relative_coords: &Point3<i32>,
+    block_start: &Point3<i32>,
+    voxel_dimensions: &Vector3<f32>,
+) -> Point3<f32> {
+    assert!(
+        !approx::relative_eq!(voxel_dimensions.x, 0.0)
+            && !approx::relative_eq!(voxel_dimensions.y, 0.0)
+            && !approx::relative_eq!(voxel_dimensions.z, 0.0),
+        "Voxel dimensions can't be 0.0"
+    );
+    Point3::new(
+        (relative_coords.x + block_start.x) as f32 * voxel_dimensions.x,
+        (relative_coords.y + block_start.y) as f32 * voxel_dimensions.y,
+        (relative_coords.z + block_start.z) as f32 * voxel_dimensions.z,
+    )
+}
+
+/// Converts relative voxel coordinates into relative voxel coordinates
+#[allow(dead_code)]
+fn absolute_to_relative_voxel_coords(
+    absolute_coords: &Point3<i32>,
+    block_start: &Point3<i32>,
+) -> Point3<i32> {
+    absolute_coords - block_start.coords
+}
+
+/// Converts absolute voxel coordinates into relative voxel coordinates
+#[allow(dead_code)]
+fn relative_to_absolute_voxel_coords(
+    relative_coords: &Point3<i32>,
+    block_start: &Point3<i32>,
+) -> Point3<i32> {
+    relative_coords + block_start.coords
+}
+
+/// Calculates the absolute coordinates of the block end
+#[allow(dead_code)]
+fn calculate_block_end(block_start: &Point3<i32>, block_dimensions: &Vector3<u32>) -> Point3<i32> {
+    Point3::new(
+        block_start.x + cast_i32(block_dimensions.x),
+        block_start.y + cast_i32(block_dimensions.y),
+        block_start.z + cast_i32(block_dimensions.z),
+    )
```

Comment:
Since these 3 are quite trivial, I'd only add them once we really need them. Otherwise we are just paying for them with compile time.

Reply:
They'll come back with the very next PR ;)